tools/toolchains/arm.py

```diff
@@ -568,6 +587,9 @@ def __init__(self, target, *args, **kwargs):
             if "--library_type=microlib" not in self.flags['asm']:
                 self.flags['asm'].append("--library_type=microlib")
 
+        if getattr(target, "default_lib", "std") == "small" and getattr(target, "ulib_supported", False) == True: 
+            self.flags['ld'].append("--predefine=-D__MICROLIB")
```

Comment:
You should include as part of the if statement above and follow the same pattern
```
if "--predefine=-D__MICROLIB" not in self.flags['ld']:
   self.flags['ld'].append("--predefine=-D__MICROLIB")
```

Reply:
We need to specify linker predefine with __MICROLIB to enable microlib related content in scatter file and this linker configuration is not required for uARM toolchain